src/main/scala/org/hammerlab/guacamole/commands/jointcaller/VCFOutput.scala

```diff
@@ -114,100 +137,244 @@ object VCFOutput {
-      (if (includePooledNormal)
-        Seq(Input(samplesEvidence.normalDNAPooledIndex, "pooled_normal", "[no path]", TissueType.Normal, Analyte.DNA))
-      else Seq.empty) ++
-      (if (includePooledTumor)
-        Seq(Input(samplesEvidence.tumorDNAPooledIndex, "pooled_tumor", "[no path]", TissueType.Tumor, Analyte.DNA))
-      else Seq.empty)
+    val effectiveInputs =
+      subInputs.items ++
+      (
+        if (includePooledNormal)
+          Seq(
+            Input(
+              samplesEvidence.normalDNAPooledIndex,
+              "pooled_normal",
+              "[no path]",
+              TissueType.Normal,
+              Analyte.DNA
+            )
+          )
+        else
+          Seq.empty
+      ) ++
+      (
+        if (includePooledTumor)
+          Seq(
+            Input(
+              samplesEvidence.tumorDNAPooledIndex,
+              "pooled_tumor",
+              "[no path]",
+              TissueType.Tumor,
+              Analyte.DNA
+            )
+          )
+        else
+          Seq.empty
+      )
 
     val genotypes = effectiveInputs.map(input => {
+
       def alleleToString(x: String) = if (x.isEmpty) "." else x
-      def mixtureToString(mixture: Map[String, Double]) = mixture.map(pair => {
-        "%s->%.2f".format(alleleToString(pair._1), pair._2)
-      }).mkString("|")
+
+      def mixtureToString(mixture: AlleleMixture) =
+        (for {
+          (allele, vaf) <- mixture
+        } yield
+          "%s->%.2f".format(alleleToString(allele), vaf)
+        ).mkString("|")
+
+      def mixturesToString(posteriors: Map[AlleleMixture, Double]): String =
+        (for {
+          (mixture, probability) <- posteriors
+        } yield
+          "[%s]=%.8g".format(
+            mixtureToString(mixture),
+            probability
+          )
+        ).mkString(",")
+
 
       val evidence = samplesEvidence.allEvidences(input.index)
+
       val genotypeBuilder = new GenotypeBuilder()
+
       genotypeBuilder.name(input.sampleName)
+
       (input.tissueType, input.analyte) match {
-        case (TissueType.Normal, Analyte.DNA) => {
-          val germlineEvidence = samplesEvidence.allEvidences(input.index).asInstanceOf[NormalDNASingleSampleSingleAlleleEvidence]
+        case (TissueType.Normal, Analyte.DNA) =>
+          val germlineEvidence =
+            samplesEvidence
+              .allEvidences(input.index)
+              .asInstanceOf[NormalDNASingleSampleSingleAlleleEvidence]
+
           val posteriors = samplesEvidence.perNormalSampleGermlinePosteriors(input.index)
+
           val alleleGenotype = posteriors.maxBy(_._2)._1
-          genotypeBuilder.alleles(JavaConversions.seqAsJavaList(
-            Seq(makeHtsjdkAllele(alleleGenotype._1), makeHtsjdkAllele(alleleGenotype._2))))
-          genotypeBuilder.attribute("RL",
-            posteriors.map(pair => "[%s/%s]=%.8g".format(pair._1._1, pair._1._2, pair._2)).mkString(","))
+
+          genotypeBuilder.alleles(
```

Comment:
the `.alleles` call in this block seems harder to read to me than what it replaces. Denser code is often clearer to my eyes I guess. The `.attribute` call below seems like an improvement though since it now names the elements in the pair, so I'm down to make that replacement. Although, just fwiw, I often find scala's `for` harder to read than alternative ways of writing the same thing, so if there's a way to do this while keeping it a `map` that would be my preference